fastly/data_source_fastly_tls_private_key.go

```diff
@@ -0,0 +1,189 @@
+package fastly
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/fastly/go-fastly/v3/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceFastlyTLSPrivateKey() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTLSPrivateKeyRead,
```

Comment:
Should we keep the CRUD function names consistent with the parent resource?

e.g.
```suggestion
		Read: dataSourceFastlyTLSPrivateKeyRead,
```